File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/entities/UserProblemStatus.java
package io.github.sanyavertolet.edukate.backend.entities;

import io.github.sanyavertolet.edukate.common.SubmissionStatus;
import lombok.Data;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Document("problem_status")
@CompoundIndex(name = "uniq_user_problem", def = "{ 'userId': 1, 'problemId': 1 }", unique = true)
public class UserProblemStatus {
    private String userId;

    private String problemId;

    private SubmissionStatus latestStatus;

    private Instant latestTime;

    private String latestSubmissionId;

    private SubmissionStatus bestStatus;

    private Instant bestTime;

    private String bestSubmissionId;
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/entities/UserProblemStatus.kt
package io.github.sanyavertolet.edukate.backend.entities

import io.github.sanyavertolet.edukate.common.SubmissionStatus
import org.springframework.data.mongodb.core.index.CompoundIndex
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document("problem_status")
@CompoundIndex(name = "uniq_user_problem", def = "{ 'userId': 1, 'problemId': 1 }", unique = true)
data class UserProblemStatus(
    val userId: String,
    val problemId: String,
    val latestStatus: SubmissionStatus,
    val latestTime: Instant,
    val latestSubmissionId: String,
    val bestStatus: SubmissionStatus,
    val bestTime: Instant,
    val bestSubmissionId: String,
)
